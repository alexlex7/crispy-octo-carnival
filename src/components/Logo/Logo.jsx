import React from 'react';
import styles from './Logo.module.css';

const defaultURL = 'http://localhost:3000/';

export default function Logo({ text = 'DudeShape', logoURL = defaultURL }) {
  return (
    <a className={styles.logo} href={logoURL}>
      {text}
    </a>
  );
}
