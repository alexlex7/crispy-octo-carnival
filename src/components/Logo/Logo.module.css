.logo {
  display: inline-block;

  font-family: 'Montserrat', 'sans-serif';
  font-size: 24px;
  font-weight: 700;
  color: #244d4d;
  text-decoration: none;

  transition: all 0.5s ease;
}

.logo:hover,
.logo:focus {
  color: #fff;
  background-color: #578a8a;
}
