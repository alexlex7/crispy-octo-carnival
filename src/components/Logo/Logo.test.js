import { render, screen } from '@testing-library/react';
import Logo from './Logo';

describe('<Logo/> spec', () => {
  test('renders the component', async () => {
    render(<Logo />);
    const linkElement = await screen.findByRole('link', {
      name: /DudeShape/i,
    });
    expect(linkElement).toBeInTheDocument();
  });

  test('renders link', async () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
