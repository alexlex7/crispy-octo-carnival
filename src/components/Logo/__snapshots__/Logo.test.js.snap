// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> spec logo renders with correct text 1`] = `
<DocumentFragment>
  <a
    class="logo"
    href="http://localhost:3000/"
  >
    DudeShape
  </a>
</DocumentFragment>
`;

exports[`<Logo/> spec renders link 1`] = `
<DocumentFragment>
  <a
    class="logo"
    href="http://localhost:3000/"
  >
    DudeShape
  </a>
</DocumentFragment>
`;
